### ISSUE TITLE
[display] bind() with unused result should produce unused code diagnostic

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#7226]
```haxe
class Main {
	public static function main() {
		Std.string.bind("");
	}
}
```

I think `Std.string.bind("")` here should produce the usual "This code has no effect" diagnostic. Maybe it needs to be marked as `@:pure` for that to work?

**Comment by Gama11** [comment]
I've tried adding the following in `displayFields.ml` line 188, but it doesn't seem to make any difference:

```ocaml
cf.cf_meta <- [Meta.Pure,[],null_pos];
```

**Comment by Simn** [comment]
I'd go further than that and say that this should be a compiler error. We already complain about "Unnamed lvalue functions are not supported", and `bind` is basically just a shortcut for that.

@ncannasse: What do you think?

**Comment by back2dos** [comment]
@Gama11 it's not pure, because `obj.method.bind(arg)` throws if `obj` is `null`. And as unsafe casts have taught us, we must deeply honor idiotic code.

What Simn suggested is on point though.

**Comment by Simn** [comment]
@ncannasse: Okay to break (see my previous comment)?

**Comment by ncannasse** [comment]
good for me

**Comment by Geokureli** [comment]
I agree with @Simn that this should result in a compiler error. On occasion, I've copy/pasted bind code to be used as a normal call and forgotten to remove the bind. I've spent way too long staring at my code wondering "why is this not being called" only to discover its only a bind call.

All the same, any unused function call should result in a compile error, not just binds, like the following:
```haxe
class Main {
	public static function main() {
		Std.string; 
	}
}
```
